mlflow/entities/trace_info.py

```diff
@@ -115,7 +115,9 @@ def to_proto(self):
     def from_proto(cls, proto) -> "TraceInfo":
         return cls(
             trace_id=proto.trace_id,
-            client_request_id=proto.client_request_id,
+            client_request_id=(
+                proto.client_request_id if proto.HasField("client_request_id") else None
+            ),
```

Comment:
While writing the unit tests for this functionality, I discovered that client_request_id, which is an optional field, is populated on TraceInfo as an empty string (rather than None) when converting a proto message that doesn't define  client_request_id into a TraceInfo instance. Made the fix here and added a test case.